lib/src/linting/rules/no-nullable-arrays.ts

```diff
@@ -0,0 +1,34 @@
+import { flatten } from "lodash";
+import { TypeKind } from "../../models/types";
+import { extractEndpointTypes } from "../../utilities/extract-endpoint-types";
+import { extractNestedUnionTypes } from "../../utilities/extract-nested-types";
+import { resolveType } from "../../verifiers/utilities/type-resolver";
+import { LintingRule } from "../rule";
+
+/**
+ * Checks that the contract doesn't define nullable arrays.
+ */
+export const noNullableArrays: LintingRule = contract => {
+  const topLevelTypes = flatten([
+    ...contract.types,
+    ...contract.endpoints.map(extractEndpointTypes)
+  ]);
+  const unionTypes = flatten(
+    topLevelTypes.map(t => extractNestedUnionTypes(t, contract.types))
```

Comment:
Why does this apply to union types only ?

Reply:
In the Spot world, just like in TypeScript, a nullable type is represented by a union of a type with `null`.

For example:
```typescript
type A = string | null;
```

This results in the type:
```
{
  kind: "union",
  types: [
    { kind: "string" },
    { kind: "null" }
  ]
}
```